### PR TITLE
use post id in get_post to enrich the post object and make it possible for multisite

### PR DESCRIPTION
**Pull request by EduardMaghakyan**
fix for #235 
Just passing post ID as argument to get_post will override some data in the initial post object and will not work on multisite, but it's better to use as it will return all possible fields. Then merging it with the initial object populates all fields and keeps the initial values.
I've added additional function to make this method work for multisite as well.

### CODE REVIEW

**Comment by tlovett1** [comment]
@EduardMaghakyan anyway you could push a test that demonstrates this bug?

File: classes/class-ep-wp-query-integration.php [hunk 1]
```diff
@@ -298,6 +311,58 @@ public function filter_posts_request( $request, $query ) {
...
+		$post_objs = array();
+		$posts_ordered = array();
+		$grouped = array();
+		$order = array();
+
+		// group by site_id to decrease number of switch_to_blogs
+		foreach ( $post_list as $post_data ) {
+			$grouped[$post_data->site_id][] = $post_data;
+			// use this to keep the initial order of posts
+			$order[] = md5( $post_data->ID . ' ' . $post_data->post_date_gmt . ' ' . $post_data->site_id );
```
Comment (tlovett1) [inline classes/class-ep-wp-query-integration.php]: Why `md5` and `post_date_gmt`? Can't we just store like this:
```php
$order[] = $post_data->ID . '-' . $post_data->site_id;
```

File: classes/class-ep-wp-query-integration.php [hunk 2]
```diff
@@ -298,6 +311,58 @@ public function filter_posts_request( $request, $query ) {
 	}
 
 	/**
+	 * Get post object data in multisite network.
+	 * Using post ID is better as get_post function pupulates all possible fields
+	 * So passing post ID and then merging with inital post object pupulates all expected fields
+	 * for WP_Post object
+	 *
+	 * @param  array $post_list list of found post objects
```
Comment (tlovett1) [inline classes/class-ep-wp-query-integration.php]: Please add @since 1.4

File: classes/class-ep-wp-query-integration.php [hunk 3]
```diff
@@ -298,6 +311,58 @@ public function filter_posts_request( $request, $query ) {
...
+	 * @return array            list of WP_Post objects
+	 */
+	public function fetch_wp_post_object( $post_list ) {
```
Comment (tlovett1) [inline classes/class-ep-wp-query-integration.php]: Rename to `fill_post_objects`

**Comment by EduardMaghakyan** [comment]
@tlovett1 I've added unit test to check the author ID, this should show the bug

File: classes/class-ep-wp-query-integration.php (hunk 1 above)
Comment (EduardMaghakyan) [inline classes/class-ep-wp-query-integration.php]: Sure we can, I've removed that part md5 and post_date_gmt

File: classes/class-ep-wp-query-integration.php (hunk 3 above)
Comment (EduardMaghakyan) [inline classes/class-ep-wp-query-integration.php]: Done

File: classes/class-ep-wp-query-integration.php (hunk 2 above)
Comment (EduardMaghakyan) [inline classes/class-ep-wp-query-integration.php]: Added

**Pull request merged**